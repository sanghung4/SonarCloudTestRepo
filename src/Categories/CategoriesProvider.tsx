import { ReactNode, createContext } from 'react';

import { Maybe, Category, useProductCategoriesQuery } from 'generated/graphql';
import { useDomainInfo } from 'hooks/useDomainInfo';

type Categories = Maybe<Array<Maybe<Category>>>;

type CategoriesContextType = {
  categories?: Categories;
  categoriesLoading: boolean;
};

export const CategoriesContext = createContext({
  categoriesLoading: false
} as CategoriesContextType);

type Props = {
  children: ReactNode;
};

function CategoriesProvider(props: Props) {
  /**
   * Data
   */
  const { engine } = useDomainInfo();
  const { data: categoriesData, loading: categoriesLoading } =
    useProductCategoriesQuery({
      skip: !engine,
      variables: { engine }
    });

  return (
    <CategoriesContext.Provider
      value={{
        categories: categoriesData?.productCategories?.categories as Categories,
        categoriesLoading
      }}
    >
      {props.children}
    </CategoriesContext.Provider>
  );
}

export default CategoriesProvider;
