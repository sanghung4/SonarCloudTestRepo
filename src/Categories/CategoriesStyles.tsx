import {
  List,
  ListItemButton,
  ListItemText,
  alpha,
  styled
} from '@dialexa/reece-component-library';

export const ListStyled = styled(List)(({ theme }) => ({
  overflowY: 'scroll',
  [theme.breakpoints.up('sm')]: {
    padding: theme.spacing(1.5, 0),
    borderRight: '1px solid #e4e4e4',
    width: '25%',
    flexShrink: 0
  }
}));

export const ListItemButtonStyled = styled(ListItemButton)(({ theme }) => ({
  [theme.breakpoints.up('sm')]: {
    marginTop: 6,
    marginBottom: 6,
    paddingTop: 4,
    paddingBottom: 4,
    '&.Mui-focusVisible': {
      backgroundColor: alpha(theme.palette.primary02.main, 0.1)
    },
    '&.Mui-selected': {
      backgroundColor: alpha(theme.palette.primary02.main, 0.1),
      borderRight: `5px solid ${theme.palette.primary02.main}`,
      '&:hover': {
        backgroundColor: alpha(theme.palette.primary02.main, 0.1)
      }
    }
  },
  '&.MuiListItemButton-gutters': {
    [theme.breakpoints.up('md')]: {
      paddingLeft: 12,
      paddingRight: 12
    },
    [theme.breakpoints.down('md')]: {
      paddingLeft: 32,
      paddingRight: 32
    }
  }
}));

export const ListItemTextStyled = styled(ListItemText)(({ theme }) => ({
  '& .MuiListItemText-primary': {
    [theme.breakpoints.down('md')]: {
      color: theme.palette.text.primary
    }
  }
}));
