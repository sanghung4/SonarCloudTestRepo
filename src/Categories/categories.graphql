query Categories($erp: String) {
  categories(erp: $erp) {
    mincronCategories
    eclipseCategories
  }
}

query ProductCategories($engine: String!) {
  productCategories(engine: $engine) {
    categories {
      name
      children {
        name
        children {
          name
        }
      }
    }
  }
}
