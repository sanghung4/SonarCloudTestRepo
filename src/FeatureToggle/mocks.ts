import { Feature } from 'generated/graphql';

export const mockFeatures: Feature[] = [
  { id: '1', isEnabled: false, name: 'noooo' },
  { id: '2', isEnabled: true, name: 'fine' },
  { id: '3', isEnabled: true, name: 'test' }
];
