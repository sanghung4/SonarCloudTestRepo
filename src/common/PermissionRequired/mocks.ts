import { UserContextType } from 'AuthProvider';
import { Permission } from 'common/PermissionRequired';

export const authConfigNoPerm = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
export const authConfig = {
  authState: {
    isAuthenticated: true
  },
  profile: {
    userId: 'test',
    permissions: [Permission.EDIT_LIST, Permission.APPROVE_CART],
    isEmployee: true,
    isVerified: true
  } as UserContextType['profile'],
  activeFeatures: ['LISTS']
};
