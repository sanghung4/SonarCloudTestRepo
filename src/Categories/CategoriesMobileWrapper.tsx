import { useContext, useState } from 'react';

import {
  Box,
  Button,
  Divider,
  IconButton,
  Link,
  Slide,
  Typography,
  useTheme
} from '@dialexa/reece-component-library';
import { useTranslation } from 'react-i18next';
import { Link as RouterLink, useHistory } from 'react-router-dom';

import { CategoriesContext } from 'Categories/CategoriesProvider';
import {
  ListStyled,
  ListItemButtonStyled,
  ListItemTextStyled
} from 'Categories/CategoriesStyles';
import Loader from 'common/Loader';
import { Category } from 'generated/graphql';
import { ChevronLeftIcon, CloseIcon } from 'icons';

type Props = {
  in: boolean;
  handleClose: (drawer: boolean) => void;
};

function CategoriesMobileWrapper(props: Props) {
  /**
   * Custom Hooks
   */
  const history = useHistory();
  const theme = useTheme();
  const { t } = useTranslation();

  /**
   * Context
   */
  const { categories, categoriesLoading } = useContext(CategoriesContext);

  /**
   * State
   */
  const [selectedCategories, setSelectedCategories] = useState<Category[]>();

  /**
   * Effects
   */

  return (
    <Slide direction="left" in={props.in}>
      <Box
        position="fixed"
        display="flex"
        flexDirection="column"
        flex="1 0 auto"
        top={0}
        left={0}
        right={0}
        height={1}
        zIndex={1}
        sx={{ bgcolor: 'common.white' }}
      >
        <Box
          mt={1.5}
          mb={1.5}
          mx={3}
          display="flex"
          justifyContent="space-between"
          alignItems="center"
        >
          <Box display="flex" flex="1" />
          <Typography
            variant="h5"
            color="primary"
            align="center"
            sx={{ fontWeight: 700 }}
          >
            {t('common.browse')}
          </Typography>
          <Box display="flex" flex="1" justifyContent="flex-end">
            <IconButton
              onClick={() => props.handleClose(true)}
              size="large"
              sx={{ p: 0 }}
            >
              <CloseIcon />
            </IconButton>
          </Box>
        </Box>
        <Divider />
        <Box pt={2} pb={3}>
          <Button
            variant="text"
            data-testid="category-back"
            startIcon={<ChevronLeftIcon />}
            onClick={() => handleBack()}
          >
            {t('common.back')}
          </Button>
        </Box>
        {categoriesLoading ? (
          <Loader />
        ) : (
          <>
            {/* CATEGORY HEADING - start */}
            <Box
              px={4}
              pb={1.5}
              display="flex"
              justifyContent="space-between"
              alignItems="flex-start"
              color="primary02.main"
            >
              <Typography
                variant="body1"
                data-testid="category-header"
                color="textPrimary"
                sx={{ pr: 2, fontWeight: 700 }}
              >
                {selectedCategories?.length
                  ? `${
                      selectedCategories[selectedCategories.length - 1].name
                    } (${
                      selectedCategories[selectedCategories.length - 1]
                        ?.children?.length
                    })`
                  : `${t('product.productCategories')} (${
                      categories?.length ?? 0
                    })`}
              </Typography>
              {selectedCategories?.length ? (
                <Link
                  color="inherit"
                  variant="body1"
                  data-testid="view-all-categories"
                  onClick={() => handleClose()}
                  component={RouterLink}
                  sx={{ flex: '0 0 auto' }}
                  to={{
                    pathname: '/search',
                    search: `categories=${selectedCategories
                      .map((c) => encodeURIComponent(c.name).trim())
                      .join('&categories=')}`
                  }}
                >
                  {t('common.viewAll')}
                </Link>
              ) : null}
            </Box>
            {/* CATEGORY HEADING - end */}
            {/* CATEGORY 1 - start */}
            <Box
              display="flex"
              flex={1}
              flexDirection="column"
              overflow="hidden"
              position="relative"
            >
              <ListStyled sx={{ zIndex: 1 }}>
                {categories?.map((category1, i) => (
                  <ListItemButtonStyled
                    key={category1?.name}
                    data-testid={`category1-${i}`}
                    onClick={() => category1 && handleCategorySelect(category1)}
                  >
                    <ListItemTextStyled primary={category1?.name} />
                  </ListItemButtonStyled>
                ))}
              </ListStyled>
              {/* CATEGORY 1 - end */}
              {/* CATEGORY 2 - start */}
              <Box
                position="absolute"
                top="0"
                left="0"
                right="0"
                bottom="0"
                zIndex={selectedCategories?.length ? 2 : 0}
                display="flex"
                flexDirection="column"
                sx={{ bgcolor: 'common.white' }}
              >
                <Slide
                  direction="left"
                  in={selectedCategories?.length === 1}
                  timeout={{
                    enter: theme.transitions.duration.enteringScreen,
                    exit: theme.transitions.duration.leavingScreen / 2
                  }}
                >
                  <ListStyled>
                    {(selectedCategories?.length
                      ? selectedCategories?.[0]?.children ?? []
                      : []
                    ).map((category, i) => (
                      <ListItemButtonStyled
                        key={category?.name}
                        data-testid={`category2-${i}`}
                        onClick={() =>
                          category && handleCategorySelect(category)
                        }
                      >
                        <ListItemTextStyled primary={category?.name} />
                      </ListItemButtonStyled>
                    ))}
                  </ListStyled>
                </Slide>
              </Box>
              {/* CATEGORY 2 - end */}
              {/* CATEGORY 3 - start */}
              <Box
                position="absolute"
                top="0"
                left="0"
                right="0"
                bottom="0"
                zIndex={selectedCategories?.length === 2 ? 3 : 0}
                display="flex"
                flexDirection="column"
                sx={{ bgcolor: 'common.white' }}
              >
                <Slide
                  direction="left"
                  in={selectedCategories?.length === 2}
                  timeout={{
                    enter: theme.transitions.duration.enteringScreen,
                    exit: theme.transitions.duration.leavingScreen / 2
                  }}
                >
                  <ListStyled>
                    {(selectedCategories?.length === 2
                      ? selectedCategories?.[1]?.children ?? []
                      : []
                    ).map((category, i) => (
                      <ListItemButtonStyled
                        key={category?.name}
                        data-testid={`category3-${i}`}
                        onClick={() =>
                          category && handleCategorySelect(category)
                        }
                      >
                        <ListItemTextStyled primary={category?.name} />
                      </ListItemButtonStyled>
                    ))}
                  </ListStyled>
                </Slide>
              </Box>
              {/* CATEGORY 3 - end */}
            </Box>
          </>
        )}
      </Box>
    </Slide>
  );

  /**
   * Event Handlers
   */

  function handleBack() {
    if (selectedCategories?.length) {
      setSelectedCategories(selectedCategories.slice(0, -1));
    } else {
      handleClose(false);
    }
  }

  /**
   *
   * @param closeDrawer Whether to close the whole drawer or just the Browse Products
   */
  function handleClose(closeDrawer: boolean = true) {
    props.handleClose(closeDrawer);

    // Wait until the drawer has closed before resetting the Category data
    setTimeout(() => {
      setSelectedCategories([]);
    }, theme.transitions.duration.leavingScreen);
  }

  function handleCategorySelect(category: Category) {
    if (category?.children?.length) {
      setSelectedCategories([...(selectedCategories ?? []), category]);
    } else {
      history.push({
        pathname: '/search',
        search: `categories=${[...(selectedCategories ?? []), category]
          .map((c) => encodeURIComponent(c?.name ?? '').trim())
          .join('&categories=')}`
      });

      handleClose();
    }
  }
}

export default CategoriesMobileWrapper;
