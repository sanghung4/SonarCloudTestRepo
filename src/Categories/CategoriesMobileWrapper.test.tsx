import { ReactNode } from 'react';
import userEvent from '@testing-library/user-event';
import { useHistory } from 'react-router-dom';
import useResizeObserver from 'use-resize-observer';

import CategoriesMobileWrapper from 'Categories/CategoriesMobileWrapper';
import { CategoriesContext } from 'Categories/CategoriesProvider';
import { success } from 'Categories/mocks';
import { Category } from 'generated/graphql';
import { render } from 'test-utils/TestWrapper';
import { setBreakpoint } from 'test-utils/mockMediaQuery';

jest.mock('use-resize-observer', () => ({
  __esModule: true,
  default: jest.fn()
}));

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: jest.fn()
}));

type MockCategoryContextProps = {
  children: ReactNode;
  data?: Category[];
  loading: boolean;
};

const MockCategoriesContext = ({
  children,
  data,
  loading
}: MockCategoryContextProps) => {
  const mockValues = {
    categories: data,
    categoriesLoading: loading
  };
  return (
    <CategoriesContext.Provider value={mockValues}>
      {children}
    </CategoriesContext.Provider>
  );
};

const mockCategoryData = success.result.data?.productCategories
  ?.categories as Category[];

describe('CategoriesButtonWrapper tests', () => {
  beforeEach(() => {
    setBreakpoint('mobile');
    (useResizeObserver as jest.Mock).mockReturnValue({
      ref: null,
      width: 1024
    });
  });

  it('should render correctly', () => {
    const { container, getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={jest.fn()} />
      </MockCategoriesContext>
    );
    expect(getByTestId('category-header')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it('should render correctly on loading', () => {
    const { container } = render(
      <MockCategoriesContext data={undefined} loading={true}>
        <CategoriesMobileWrapper in={true} handleClose={jest.fn()} />
      </MockCategoriesContext>
    );
    expect(container).toMatchSnapshot();
  });

  it('should navigate through the categories', () => {
    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={jest.fn()} />
      </MockCategoriesContext>
    );
    const headerSpan = getByTestId('category-header');
    userEvent.click(getByTestId('category1-2'));
    expect(headerSpan).toHaveTextContent('Pipe & Fittings (5)');

    userEvent.click(getByTestId('category2-1'));
    expect(headerSpan).toHaveTextContent(
      'Plastic Pipe & Fittings - Supply (2)'
    );
  });

  it('should navigate back through the categories', () => {
    const handleClose = jest.fn();

    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={handleClose} />
      </MockCategoriesContext>
    );
    const headerSpan = getByTestId('category-header');
    const backButton = getByTestId('category-back');

    userEvent.click(getByTestId('category1-2'));
    userEvent.click(getByTestId('category2-1'));
    userEvent.click(backButton);
    expect(headerSpan).toHaveTextContent('Pipe & Fittings (5)');

    userEvent.click(backButton);
    expect(headerSpan).toHaveTextContent('Product Categories (8)');

    userEvent.click(backButton);
    expect(handleClose).toHaveBeenCalledTimes(1);
  });

  it('should link to the category from the view all button', () => {
    const handleClose = jest.fn();

    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={handleClose} />
      </MockCategoriesContext>
    );

    // Navigate to bottom category
    userEvent.click(getByTestId('category1-2'));
    const viewAllButton = getByTestId('view-all-categories');
    expect(viewAllButton.getAttribute('href')).toEqual(
      '/search?categories=Pipe%20%26%20Fittings'
    );

    userEvent.click(getByTestId('category2-0'));
    expect(viewAllButton.getAttribute('href')).toEqual(
      '/search?categories=Pipe%20%26%20Fittings&categories=Metal%20Pipe%20%26%20Fittings'
    );

    userEvent.click(viewAllButton);
    expect(handleClose).toHaveBeenCalled();
  });

  it('should navigate and close the sidebar if a category three is selected', () => {
    const push = jest.fn();
    const handleClose = jest.fn();

    (useHistory as jest.Mock).mockReturnValue({
      push
    });

    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={handleClose} />
      </MockCategoriesContext>
    );

    // Navigate to bottom category
    userEvent.click(getByTestId('category1-0'));
    userEvent.click(getByTestId('category2-2'));
    userEvent.click(getByTestId('category3-2'));

    expect(handleClose).toHaveBeenCalled();
    expect(push).toHaveBeenCalledWith({
      pathname: '/search',
      search:
        'categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Appliances&categories=Outdoor%20Grills%20and%20BBQ'
    });
  });

  it('should navigate and close the sidebar if a category two is selected that has no children', () => {
    const push = jest.fn();
    const handleClose = jest.fn();

    (useHistory as jest.Mock).mockReturnValue({
      push
    });

    const { getByTestId } = render(
      <MockCategoriesContext data={mockCategoryData} loading={false}>
        <CategoriesMobileWrapper in={true} handleClose={handleClose} />
      </MockCategoriesContext>
    );

    // Navigate to bottom category
    userEvent.click(getByTestId('category1-0'));
    userEvent.click(getByTestId('category2-5'));

    expect(handleClose).toHaveBeenCalled();
    expect(push).toHaveBeenCalledWith({
      pathname: '/search',
      search:
        'categories=Faucets%2C%20Fixtures%20%26%20Appliances&categories=Tools'
    });
  });
});
