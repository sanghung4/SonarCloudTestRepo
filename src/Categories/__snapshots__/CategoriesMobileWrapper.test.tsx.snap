// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesButtonWrapper tests should render correctly 1`] = `
<div>
  <div
    class="MuiBox-root css-1nnf5qx"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiBox-root css-1lm7c2e"
    >
      <div
        class="MuiBox-root css-13pmxen"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-wc1ui5-MuiTypography-root"
      >
        Browse
      </h5>
      <div
        class="MuiBox-root css-1xfrdmu"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1yy1t52-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg>
            delete.svg
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-ekan51"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-lw1dhn-MuiButtonBase-root-MuiButton-root"
        data-testid="category-back"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
        >
          <svg>
            chevron-left.svg
          </svg>
        </span>
        Back
      </button>
    </div>
    <div
      class="MuiBox-root css-s2j4kj"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1rcon39-MuiTypography-root"
        data-testid="category-header"
      >
        Product Categories (8)
      </p>
    </div>
    <div
      class="MuiBox-root css-6ezlc4"
    >
      <ul
        class="MuiList-root css-1kf6xy5-MuiList-root"
      >
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-0"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Faucets, Fixtures & Appliances
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-1"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Plumbing Installation, Tools, Hardware & Safety
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-2"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Pipe & Fittings
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-3"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Heating & Cooling
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-4"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Hot Water, Valves, Irrigation & Pumps
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-5"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              TBC
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-6"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Faucets, Fixtures Appliances
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <div
          class="MuiListItemButton-root MuiListItemButton-gutters MuiButtonBase-root css-hrv000-MuiButtonBase-root-MuiListItemButton-root"
          data-testid="category1-7"
          role="button"
          tabindex="0"
        >
          <div
            class="MuiListItemText-root css-1y7mhd5-MuiListItemText-root"
          >
            <span
              class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-zvlsvz-MuiTypography-root"
            >
              Faucets, FiXtures & Appliances
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </ul>
      <div
        class="MuiBox-root css-8zx45r"
      >
        <ul
          class="MuiList-root css-172nema-MuiList-root"
          style="visibility: hidden; webkit-transform: translateX(414px); transform: translateX(414px);"
        />
      </div>
      <div
        class="MuiBox-root css-8zx45r"
      >
        <ul
          class="MuiList-root css-172nema-MuiList-root"
          style="visibility: hidden; webkit-transform: translateX(414px); transform: translateX(414px);"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CategoriesButtonWrapper tests should render correctly on loading 1`] = `
<div>
  <div
    class="MuiBox-root css-1nnf5qx"
    style="webkit-transform: none; transform: none; webkit-transition: -webkit-transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms; transition: transform 225ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;"
  >
    <div
      class="MuiBox-root css-1lm7c2e"
    >
      <div
        class="MuiBox-root css-13pmxen"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-wc1ui5-MuiTypography-root"
      >
        Browse
      </h5>
      <div
        class="MuiBox-root css-1xfrdmu"
      >
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeLarge css-1yy1t52-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg>
            delete.svg
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-di86mu-MuiDivider-root"
    />
    <div
      class="MuiBox-root css-ekan51"
    >
      <button
        class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButtonBase-root  css-lw1dhn-MuiButtonBase-root-MuiButton-root"
        data-testid="category-back"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1sx3uk1-MuiButton-startIcon"
        >
          <svg>
            chevron-left.svg
          </svg>
        </span>
        Back
      </button>
    </div>
    <div
      class="MuiBox-root css-1nen2z4"
      data-testid="loader-component"
    >
      <div
        class="MuiBox-root css-hyxlzm"
      >
        <span
          aria-valuenow="100"
          class="MuiCircularProgress-root MuiCircularProgress-determinate MuiCircularProgress-colorPrimary css-1wyqyef-MuiCircularProgress-root"
          role="progressbar"
          style="width: 80px; height: 80px; transform: rotate(-90deg);"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleDeterminate css-oxts8u-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="19.5"
              stroke-width="5"
              style="stroke-dasharray: 122.522; stroke-dashoffset: 0.000px;"
            />
          </svg>
        </span>
        <span
          class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-1gse1c1-MuiCircularProgress-root"
          role="progressbar"
          style="width: 80px; height: 80px;"
        >
          <svg
            class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
            viewBox="22 22 44 44"
          >
            <circle
              class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
              cx="44"
              cy="44"
              fill="none"
              r="19.5"
              stroke-width="5"
            />
          </svg>
        </span>
      </div>
    </div>
  </div>
</div>
`;
